### ISSUE TITLE
Error when testing interface.pp

### ISSUE THREAD

**Issue by Cosaquee** [coi-gov-pl/puppet-jboss#50]
```
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > ensure
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > exists?
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > compiledpath
Debug: Command send to JBoss CLI: /interface=public-additional:read-resource(include-runtime=true, include-defaults=false)
Debug: Output from JBoss CLI [#<Process::Status: pid=3890,exited(1)>]: {
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[(\"interface\" => \"public-additional\")]' not found",
    "rolled-back" => true
}

Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > OUT > status(): false
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > OUT > ensure(): :absent
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > ensure=(:present)
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > create
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > exists?
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > compiledpath
Debug: Puppet::Type::Jboss_confignode::ProviderJbosscli[/interface=public-additional] > IN > compileprops
Debug: Command send to JBoss CLI: /interface=public-additional:add(up=nil, nic=nil, site-local-address=nil, public-address=nil, any-address=nil, any-ipv4-address=nil, any-ipv6-address=nil, virtual=nil, subnet-match=nil, inet-address=nil, point-to-point=nil, nic-match=nil, loopback-address=nil, link-local-address=nil, multicast=nil, loopback=nil)
Warning: JBoss CLI command failed, try 1/3, last status: 1, message: {
    "outcome" => "failed",
    "failure-description" => "JBAS014690: any-address is invalid",
    "rolled-back" => true
}

Debug: Command send to JBoss CLI: /interface=public-additional:add(up=nil, nic=nil, site-local-address=nil, public-address=nil, any-address=nil, any-ipv4-address=nil, any-ipv6-address=nil, virtual=nil, subnet-match=nil, inet-address=nil, point-to-point=nil, nic-match=nil, loopback-address=nil, link-local-address=nil, multicast=nil, loopback=nil)
Warning: JBoss CLI command failed, try 2/3, last status: 1, message: {
    "outcome" => "failed",
    "failure-description" => "JBAS014690: any-address is invalid",
    "rolled-back" => true
}

Debug: Command send to JBoss CLI: /interface=public-additional:add(up=nil, nic=nil, site-local-address=nil, public-address=nil, any-address=nil, any-ipv4-address=nil, any-ipv6-address=nil, virtual=nil, subnet-match=nil, inet-address=nil, point-to-point=nil, nic-match=nil, loopback-address=nil, link-local-address=nil, multicast=nil, loopback=nil)
Warning: JBoss CLI command failed, try 3/3, last status: 1, message: {
    "outcome" => "failed",
    "failure-description" => "JBAS014690: any-address is invalid",
    "rolled-back" => true
}

Debug: Command send to JBoss CLI: /interface=public-additional:add(up=nil, nic=nil, site-local-address=nil, public-address=nil, any-address=nil, any-ipv4-address=nil, any-ipv6-address=nil, virtual=nil, subnet-match=nil, inet-address=nil, point-to-point=nil, nic-match=nil, loopback-address=nil, link-local-address=nil, multicast=nil, loopback=nil)
Debug: Output from JBoss CLI [#<Process::Status: pid=4041,exited(1)>]: {
    "outcome" => "failed",
    "failure-description" => "JBAS014690: any-address is invalid",
    "rolled-back" => true
}

Error:
Configuration node failed to create:
[CLI command]: /interface=public-additional:add(up=nil, nic=nil, site-local-address=nil, public-address=nil, any-address=nil, any-ipv4-address=nil, any-ipv6-address=nil, virtual=nil, subnet-match=nil, inet-address=nil, point-to-point=nil, nic-match=nil, loopback-address=nil, link-local-address=nil, multicast=nil, loopback=nil)
[Error message]: {
    "outcome" => "failed",
    "failure-description" => "JBAS014690: any-address is invalid",
    "rolled-back" => true
}

/etc/puppet/modules/jboss/lib/puppet/provider/jbosscli.rb:229:in `executeWithFail'
/etc/puppet/modules/jboss/lib/puppet/provider/jbosscli.rb:188:in `bringUp'
/etc/puppet/modules/jboss/lib/puppet/provider/jboss_confignode/jbosscli.rb:14:in `create'
/etc/puppet/modules/jboss/lib/puppet/provider/jboss_confignode/jbosscli.rb:126:in `ensure='
/usr/lib/ruby/site_ruby/1.8/puppet/property.rb:178:in `send'
/usr/lib/ruby/site_ruby/1.8/puppet/property.rb:178:in `call_provider'
/usr/lib/ruby/site_ruby/1.8/puppet/property.rb:503:in `set'
/usr/lib/ruby/site_ruby/1.8/puppet/property.rb:581:in `sync'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction/resource_harness.rb:204:in `sync'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction/resource_harness.rb:128:in `sync_if_needed'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction/resource_harness.rb:81:in `perform_changes'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction/resource_harness.rb:20:in `evaluate'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction.rb:204:in `apply'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction.rb:217:in `eval_resource'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction.rb:147:in `call'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction.rb:147:in `evaluate'
/usr/lib/ruby/site_ruby/1.8/puppet/util.rb:327:in `thinmark'
/usr/lib/ruby/1.8/benchmark.rb:308:in `realtime'
/usr/lib/ruby/site_ruby/1.8/puppet/util.rb:326:in `thinmark'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction.rb:147:in `evaluate'
/usr/lib/ruby/site_ruby/1.8/puppet/graph/relationship_graph.rb:118:in `traverse'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction.rb:138:in `evaluate'
/usr/lib/ruby/site_ruby/1.8/puppet/resource/catalog.rb:169:in `apply'
/usr/lib/ruby/site_ruby/1.8/puppet/util/log.rb:149:in `with_destination'
/usr/lib/ruby/site_ruby/1.8/puppet/transaction/report.rb:112:in `as_logging_destination'
/usr/lib/ruby/site_ruby/1.8/puppet/resource/catalog.rb:168:in `apply'
/usr/lib/ruby/site_ruby/1.8/puppet/configurer.rb:120:in `apply_catalog'
/usr/lib/ruby/site_ruby/1.8/puppet/util.rb:161:in `benchmark'
/usr/lib/ruby/1.8/benchmark.rb:308:in `realtime'
/usr/lib/ruby/site_ruby/1.8/puppet/util.rb:160:in `benchmark'
/usr/lib/ruby/site_ruby/1.8/puppet/configurer.rb:119:in `apply_catalog'
/usr/lib/ruby/site_ruby/1.8/puppet/configurer.rb:227:in `run_internal'
/usr/lib/ruby/site_ruby/1.8/puppet/configurer.rb:134:in `run'
/usr/lib/ruby/site_ruby/1.8/puppet/context.rb:64:in `override'
/usr/lib/ruby/site_ruby/1.8/puppet.rb:244:in `override'
/usr/lib/ruby/site_ruby/1.8/puppet/configurer.rb:133:in `run'
/usr/lib/ruby/site_ruby/1.8/puppet/application/apply.rb:297:in `apply_catalog'
/usr/lib/ruby/site_ruby/1.8/puppet/application/apply.rb:231:in `main'
/usr/lib/ruby/site_ruby/1.8/puppet/context.rb:64:in `override'
/usr/lib/ruby/site_ruby/1.8/puppet.rb:244:in `override'
/usr/lib/ruby/site_ruby/1.8/puppet/application/apply.rb:193:in `main'
/usr/lib/ruby/site_ruby/1.8/puppet/application/apply.rb:154:in `run_command'
/usr/lib/ruby/site_ruby/1.8/puppet/application.rb:381:in `run'
/usr/lib/ruby/site_ruby/1.8/puppet/application.rb:507:in `plugin_hook'
/usr/lib/ruby/site_ruby/1.8/puppet/application.rb:381:in `run'
/usr/lib/ruby/site_ruby/1.8/puppet/util.rb:488:in `exit_on_fail'
/usr/lib/ruby/site_ruby/1.8/puppet/application.rb:381:in `run'
/usr/lib/ruby/site_ruby/1.8/puppet/util/command_line.rb:146:in `run'
/usr/lib/ruby/site_ruby/1.8/puppet/util/command_line.rb:92:in `execute'
/usr/bin/puppet:8
Error: /Stage[main]/Main/Jboss::Interface[public-additional]/Jboss::Clientry[/interface=public-additional]/Jboss_confignode[/interface=public-additional]/ensure: change from absent to present failed:
Configuration node failed to create:
[CLI command]: /interface=public-additional:add(up=nil, nic=nil, site-local-address=nil, public-address=nil, any-address=nil, any-ipv4-address=nil, any-ipv6-address=nil, virtual=nil, subnet-match=nil, inet-address=nil, point-to-point=nil, nic-match=nil, loopback-address=nil, link-local-address=nil, multicast=nil, loopback=nil)
[Error message]: {
    "outcome" => "failed",
    "failure-description" => "JBAS014690: any-address is invalid",
    "rolled-back" => true
}

```

When making additional tests before release I found error in tests for interfaces.pp. This need to be resolved asap.

**Comment by cardil** [comment]
Add `.reject { |k,v| v.nil? }` to parameters in create method of clientry provider. Check again and if it helps do commit to #10  Maybe a test with nil values for create method. Will you do it @Cosaquee ?

**Comment by cardil** [comment]
I meant PR #46

**Comment by cardil** [comment]
This is false positive bug report, see b280c37 for resolution (will be merged as part of PR #46).

Closing :smiley:
